apps/prairielearn/src/lib/authzData.types.ts

```diff
@@ -78,5 +78,52 @@ export const FullAuthzDataSchema = z.object({
   permissions_course_instance: SprocAuthzCourseInstanceSchema,
 });
 
-/** The user facing version that is obtained via the page context helpers. */
+export type FullAuthzData = z.infer<typeof FullAuthzDataSchema>;
+
+export type CourseInstanceRole =
+  | 'System'
+  | 'None'
+  | 'Student'
+  | 'Student Data Viewer'
+  | 'Student Data Editor'
+  // The role 'Any' is equivalent to 'Student' OR 'Student Data Viewer' OR 'Student Data Editor'
+  | 'Any';
+
 export type AuthzData = RawPageAuthzData | DangerousSystemAuthzData;
+
+type Prettify<T> = {
+  [K in keyof T]: T[K];
+} & {}; // Force typescript to resolve the type immediately
```

Comment:
`lib/types.ts` exists for this kind of utility type; consider moving it there? While you're at it, you can make this comment a JSDoc comment for Intellisense hover.